src/nvim/ops.c

```diff
@@ -3386,6 +3387,10 @@ void do_put(int regname, yankreg_T *reg, int dir, long count, int flags)
           }
           STRMOVE(ptr, oldp + col);
           ml_replace(lnum, newp, false);
+
+          // compute the byte offset for the last character
+          first_byte_off = mb_head_off(newp, ptr - 1);
```

Comment:
```suggestion
          first_byte_off = utf_head_off(newp, ptr - 1);
```

Reply:
Thank you! I've had trouble with commit suggestions in PRs generated via `vim-patch.sh`, so I made those changes manually.